web/shapeworks/src/store/methods.ts

```diff
@@ -145,6 +146,14 @@ export async function spawnJob(action: string, payload: Record<string, any>): Pr
     return undefined;
 }
 
+export async function spawnAnalysisJob(params: {range: number, steps: number}): Promise<any> {
+    if (selectedProject.value) {
+        return (await analyzeProject(selectedProject.value.id, params))?.data
+    } else {
+        return undefined;
+    }
+}
+
```

Comment:
It might be better design to combine this method with the `spawnJob` method right above it. You can just add a `case 'analyze'` and use the payload arg for your params.